### DISCUSSION TITLE
How do you push changes to your GitHub page without deleting everything and re-uploading it?

### DISCUSSION THREAD

**Discussion by SphericalSilver** [community/community#23041] [Other Feature Feedback, Questions, & Ideas]
<p>As the title hints, I’m a complete noob at git.</p>
<p>Right now the way I’ve been updating my github page site is by deleting the repository, then making a new one with the new website contents.</p>
<p>I figure there has to be a more efficient way to just push the changes there, but I can’t figure out how to do it.</p>

**Comment by SethClydesdale** [comment, marked as answer]
<p>Hello <a class="mention" href="/sphericalsilver">@sphericalsilver</a>,</p>
<p>You want to update the master branch, correct? If so, you should be able to run  <strong>git push origin master</strong> to push your local changes to your online repository. The following page has more info on <strong>git push</strong>.</p>
<p><a href="https://help.github.com/en/articles/pushing-commits-to-a-remote-repository" rel="nofollow noopener">https://help.github.com/en/articles/pushing-commits-to-a-remote-repository</a></p>
<p>Another alternative is to use <a href="https://desktop.github.com/" rel="nofollow noopener">GitHub Desktop</a>. With this you can clone your repository to your local system, make changes, and then commit and push these changes to your online repository. This the method I use to update my GitHub pages.</p>

**Reply by abdulmukit-n** [reply]
I followed this by updating my master branch with the new changes but the changes did not reflect onto my github page(deployed state) , am i doing something wrong ?

**Reply by airtower-luna** [reply]
Maybe it just takes a little while, maybe there's some problem that prevents the deployment. It's impossible to say without seeing the repository.

However that's a separate question that should be in a separate thread. If you create one make sure to include a link to the repository so people can take a look. :wink:

**Reply by Felix404-student** [reply]
Just to back airtower-luna up, for me, changes went 'live' to the page in ~5 minutes, but not instantly. Take a deep breath, get coffee, then refresh the page.

**Comment by SphericalSilver** [comment]
<p>Thank you, that’s really helpful!</p>
